h2/src/main/org/h2/command/ddl/CreateView.java

```diff
@@ -100,7 +100,8 @@ public int update() {
             querySQL = select.getPlanSQL();
         }
         // The view creates a Prepared command object, which belongs to a
-        // session, so we pass the system session down.
+        // session, so we pass the system session (not the current session) down.
+        // Why is the SYS session used here ? I think it might cause a problem...
```

Comment:
if we are creating a view, we need to make it immediately visible to other sessions, which means we need to commit() the session we use to create it on, which is why we use the system session.
We can't commit the current session, because the application might not be ready to commit yet